onnxruntime/core/graph/graph_utils.h

```diff
@@ -86,20 +81,63 @@ bool GetRepeatedNodeAttributeValues(const Node& node,
   return false;
 }
 
-/** Removes the given Node from the Graph and keeps Graph consistent by rebuilding needed connections.
-    We support the removal of the Node as long as the following conditions hold:
-    - There should be no implicit inputs.
-    - Only one of the outputs is used by downstream operators (but it can have multiple output edges).
-    - If the Node has a single incoming node (and possibly multiple initializers), we can remove the Node and
-      connect its incoming node to its outgoing nodes.
-    - If the Node has a single initializer as input, we remove the Node and feed the initializer as input to its
-      output nodes. */
-bool RemoveNode(Graph& graph, Node& node);
+/** Tests if we can remove a node and merge its input edge (if any) with its output edges.
+Conditions:
+ - Only one of the node's outputs is used by downstream operators
+   - multiple edges for the single used output are allowed
+ - The node must have zero or one input edges 
+   - we merge any input edge with the output edges for the single output that is used
+ - If the node has no input edges it must have a single input
```

Comment:
nit: Let's make this a sub-bullet (in the bullet you say that the node must have 0 or 1 input edges. Then in the previous and this sub-bullet you analyze each case of 0 and 1.

Reply:
Edited to try and clarify

---
In reply to: [328781720](https://github.com/microsoft/onnxruntime/pull/1407#discussion_r328781720) [](ancestors = 328781720)